mailgun-relayery/src/servers/control/conversion_test.go

```diff
@@ -0,0 +1,153 @@
+package control
+
+import (
+	"servers/protoed_channel"
+	"testing"
+)
+
+func TestJSONToProto(t *testing.T) {
+	name1, name2, name3, name4 := "Ludwig van Beethoven", "Johannes Brahms", "Richard Wagner", "Robert Schumann"
+	sender := Entity{Name: &name1, Email: "ludwig.van.beethoven@composers.com"}
+	recipients := []Entity{{Name: &name2, Email: "johannes.brahms@composers.com"}}
+	cc := []Entity{{Name: &name3, Email: "richard.wagner@composers.com"}}
+	bcc := []Entity{{Name: &name4, Email: "robert.schumann@composers.com"}}
+
+	jsonChan := Channel{Descriptor: Descriptor("some-channel"),
+		Token:  Token("oqiwdJKNsdKIUwezd92DNQsndkDERDFKJNQWSwq3rODIU"),
+		Sender: sender, Recipients: recipients, Cc: cc, Bcc: bcc, MinPeriod: 0.0001, MaxSize: 10000000}
+	converted := JSONToProto(jsonChan)
+
+	if jsonChan.Descriptor != Descriptor(converted.Descriptor_) {
```

Comment:
It feels a bit strange to rewrite the entire code in the postconditions. Should I even include calls to conversion functions like `protoToJSONEntityList`? That seems an overkill to me, and a lot of repeated work for the code to carry.

Reply:
I think I figured out an easier way to unit test and QA this. What about writing a whole struct as JSON (stored in a const string as jsonText) and protobuf in text format (stored in a const string as protoText), then:
* unmarshall `jsonText` into `chan`, convert to protobuf representation, marshal protobuf representtion into text, compare the text against `protoText`
* the reverse, start with unmarshalling `protoText` into a protobuf representation

You need to implement some sort of `dedent` function (to left-strip common whitespace) so that multiple pairs of such test cases can be easily defined in an array.